components/camera/client.go

```diff
@@ -31,15 +31,13 @@ import (
 type client struct {
 	resource.Named
 	resource.TriviallyReconfigurable
-	resource.TriviallyCloseable
 	mu                      sync.Mutex
 	name                    string
 	conn                    rpc.ClientConn
 	client                  pb.CameraServiceClient
 	logger                  logging.Logger
 	activeBackgroundWorkers sync.WaitGroup
-	cancelCtx               context.Context
-	cancel                  func()
+	healthyClientCh         chan struct{}
```

Comment:
Until the resource graph can offer a better story to the resources it manages, I think we're stuck with verbosely documenting how this is intended to work. The solution we converged on only makes sense with the context that the alternatives are hard. A strawman:
```
RSDK-6340: The resource manager currently behaves such that remote resources will be closed when the underlying connection goes bad. However, when the connection is re-established, the client objects these resources represent are not re-initialized/marked "healthy". `healthyClientCh` helps track these transitions between healthy and unhealthy states.

When a new `client.Stream()` is created, we will either:
- Use the existing `healthyClientCh` or
- Create a new one to use

The goroutine a `Stream()` method spins off will listen to its version of the `healthyClientCh` to be notified when the connection has died so it can gracefully terminate.

When a connection becomes unhealthy, the resource manager will call `Close` on the camera client object. Closing the client will:
- close its `client.healthyClientCh` channel
- wait for existing "stream" goroutines to drain and
- nil out the `client.healthyClientCh` member variable

New streams concurrent with closing cannot start until this drain completes. There will never be stream goroutines from the old "generation" running concurrently with those from the new "generation".
```

Reply:
added https://github.com/viamrobotics/rdk/pull/3467/commits/cbbcdac1acc14ff91acb9a416031042accd615b7